.github/workflows/ci.yml

```diff
@@ -48,7 +48,9 @@ jobs:
       - uses: actions/checkout@v4
         with:
           submodules: "recursive"
-      - uses: spiraldb/actions/.github/actions/setup-uv@0.7.0
```

Comment:
this is a very funny back and forth. I would prefer we kept these centralized somehow

Reply:
I agree, but I just don't think we need the dedicated action. Enabling the cache here is a minor nicety IMO, and now all `uv` invocations are idempotent and don't require the `uv sync` step.

Reply:
I think my main point is that we should just never have to do a an explicit `sync`, we can always just `uv run --isolate --all-packages` which gives us a clean slate venv with all required dependencies

Reply:
In the name of moving this forward, I reverted this change for now.

Reply:
ok, sounds like we should change the wrapper action we have? Maybe we don't need it at all?

Reply:
I think so, but I also don't fully understand the impact of the env variables we set in it on maturin, and there isn't really much of a difference either way IMO.